ssl/record/rec_layer_s3.c

```diff
@@ -306,7 +306,7 @@ int ssl3_read_n(SSL *s, int n, int max, int extend, int clearold)
             if (s->mode & SSL_MODE_RELEASE_BUFFERS && !SSL_IS_DTLS(s))
                 if (len + left == 0)
                     ssl3_release_read_buffer(s);
-            return (i);
+            return -1;
```

Comment:
It's not at all obvious reading the SSL_read() manpage what the difference between 0 and -1 is. And you turned that into just 0 for SSL_read_ex(). I think 0 means end of file and that -1 means that it can be retried (like read()). But the description could clearly be improved.

But I'd really like to see at least return values documented for each function, and it's fine if you point to some other function that documents it.